legal-api/src/legal_api/models/business.py

```diff
@@ -561,6 +571,68 @@ def get_filing_by_id(cls, business_identifier: int, filing_id: str):
             one_or_none()
         return None if not filing else filing[1]
 
+    def get_alternate_names(self) -> dict:
+        """Get alternate names for this business.
+
+        - Return name translation (alias table) entries if any
+        - Return SP DBA entries in alternate name entries if not SP
+        - For SP or GP, also return an alternate name from existing business record
+        - Return empty list if there are no alternate name entries
+        """
+        alternate_names = []
+
+        for alias in self.aliases:
+            alternate_names.append(
+                {
+                    'name': alias.alias,
+                    'startDate': LegislationDatetime.format_as_legislation_date(self.founding_date),
```

Comment:
I thought `aliases` had a start date field for some reason.

We'll need to get the start date value from filing.effective_date that's tied to the transaction that created the alias entry.

Included an example where there is more than one versioning record for a given alias entry.  But basically we just need to check that the `end_transaction_id` is null to ensure we are looking at the right transaction & filing.

![image](https://github.com/bcgov/lear/assets/6685223/ee8ff2cc-c712-4221-ad09-25c4b1bde2d6)

Reply:
I've updated the start date to grab it from the filing associated with the alias entry creation